streaming/java/streaming-runtime/src/main/java/io/ray/streaming/runtime/master/coordinator/CheckpointCoordinator.java

```
@@ -0,0 +1,214 @@
+          LOG.error("Ignore interrupt checkpoint exception in catch block.");
+        }
+      }
+    }
+    LOG.warn("Checkpoint coordinator thread exit.");
+  }
+
+  public Boolean reportJobWorkerCommit(WorkerCommitReport report) {
+    LOG.info("Report job worker commit {}.", report);
+
+    Boolean ret = runtimeContext.cpCmds.offer(report);
+    if (!ret) {
+      LOG.warn("Report job worker commit failed, because command queue is full.");
+    }
+    return ret;
+  }
+
+  private void processCommitReport(WorkerCommitReport commitReport) {
+    LOG.info("Start process commit report {}, from actor name={}.", commitReport,
+        graphManager.getExecutionGraph().getActorName(commitReport.fromActorId));
+
+    try {
+      Preconditions.checkArgument(
+          commitReport.commitCheckpointId == runtimeContext.lastCheckpointId,
+          "expect checkpointId %s, but got %s",
+          runtimeContext.lastCheckpointId, commitReport);
+
+      if (!pendingCheckpointActors.contains(commitReport.fromActorId)) {
+        LOG.warn("Invalid commit report, skipped.");
+        return;
+      }
+
+      pendingCheckpointActors.remove(commitReport.fromActorId);
+      LOG.info("Pending actors after this commit: {}.",
+          graphManager.getExecutionGraph().getActorName(pendingCheckpointActors));
+
+      // checkpoint finish
+      if (pendingCheckpointActors.isEmpty()) {
+        // actor finish
+        runtimeContext.checkpointIds.add(runtimeContext.lastCheckpointId);
+
+        if (clearExpiredCpStateAndQueueMsg()) {
+          // save master context
+          jobMaster.saveContext();
+
+          LOG.info("Finish checkpoint: {}.", runtimeContext.lastCheckpointId);
+        } else {
+          LOG.warn("Fail to do checkpoint: {}.", runtimeContext.lastCheckpointId);
+        }
+      }
+
+      LOG.info("Process commit report {} success.", commitReport);
+    } catch (Throwable e) {
+      LOG.warn("Process commit report has exception.", e);
```

What should worker do when there is an exception. Ignoring it is expectable behaviour?

---

Pelase check swallowed exception and see if there is any exception we swallowed by mistake

---

yes, just ignore it and wait for the next round.